### PR TITLE
fix/passwordless-authparams-priority

### PR DESCRIPTION
**Pull request by MathiasGr**
### Changes

- Updated authParams priority for the `WebAuth` `passwordlessStart` method
- Updated relevant test

### References

Closes issue #1018 

### Testing

- [x] This change adds unit test coverage

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [x] All tests and linters described in the [Develop section](https://github.com/auth0/auth0.js#develop) run without errors

### CODE REVIEW

File: src/authentication/passwordless-authentication.js [hunk 1]
```diff
@@ -133,17 +133,19 @@ PasswordlessAuthentication.prototype.start = function(options, cb) {
 
   if (body.scope) {
     body.authParams = body.authParams || {};
-    body.authParams.scope = body.scope;
+    body.authParams.scope = body.authParams.scope || body.scope;
```
Comment (luisrudge) [inline src/authentication/passwordless-authentication.js]: don't change this line

File: src/authentication/passwordless-authentication.js [hunk 2]
```diff
@@ -133,17 +133,19 @@ PasswordlessAuthentication.prototype.start = function(options, cb) {
...
   if (body.redirectUri) {
     body.authParams = body.authParams || {};
-    body.authParams.redirect_uri = body.redirectUri;
+    body.authParams.redirect_uri =
+      body.authParams.redirectUri || body.redirectUri;
   }
 
   if (body.responseType) {
     body.authParams = body.authParams || {};
-    body.authParams.response_type = body.responseType;
```
Comment (luisrudge) [inline src/authentication/passwordless-authentication.js]: don't change this line

**Comment by MathiasGr** [comment]
@luisrudge @lbalmaceda 
I will address your comments here:

### About fixing responseType and scope behavior
[It is explicitly written in your api that these parameters can be overriden by putting them in authParams](https://auth0.com/docs/api/authentication#passwordless) so there is absolutely not reason not to fix them.

> authParams | Use this to append or override the link parameters (like scope, redirect_uri, protocol, response_type), when you send a link using email.

### About refactoring the code
Hey man I'm just a nice guy trying to help (not your refactoring intern). The common practice in open source is to only fix the issue while keeping the original structure because you usually don't have enough knowledge of the code base, and you don't have time either. So the way I see it you have two options: either you ask people to help fix your code (like @luisrudge did in the issue) and you accept these contributions, or you don't and refactor whatever you want but you can't have both !

**Comment by luisrudge** [comment]
@MathiasGr gotcha. I got confused with the overrides, but you're right about the others as well. Would you mind adding the extra params (`scope` and `responseType`) to the override test case so we don't reintroduce this bugs in the future? Thanks for the help!

**Comment by MathiasGr** [comment]
Done ! I also added `protocol` since it's part of the params listed in [the docs](https://auth0.com/docs/api/authentication#get-code-or-link).

**Comment by luisrudge** [comment]
Thanks for the PR @MathiasGr! There's a code coverage issue, but I'll fix this tomorrow 🎉

**Comment by MathiasGr** [comment]
Hi there, anything wrong with the PR ? Let me know...

**Comment by stevehobbsdev** [comment]
Closing this PR in favour of #1058, which fixes the coverage issue.

**Pull request closed without merging**